tests/system_tests/test_core/test_metric_full.py

```diff
@@ -4,211 +4,126 @@
 import wandb
 
 
-@pytest.mark.wandb_core_only
-def test_mode_shared(user, relay_server, test_settings):
-    # test that history records are properly flushed
-    metrics = [f"metric_{i}" for i in range(2)]
-
-    with relay_server() as relay:
-        run = wandb.init(settings=test_settings({"mode": "shared"}))
-        run.define_metric(name="metric_*", step_metric="train_step")
-
-        for train_step in range(3):
-            for metric in metrics:
-                run.log(
-                    {
-                        "train_step": train_step,
-                        metric: train_step * 2,
-                    }
-                )
-
-        run.finish()
-
-    # assert that relay.context.history["metric_i"] both have three NaN values
-    # because a history update should look like {"train_step": <step>, "metric_i": <value>}
-    for metric in metrics:
-        assert relay.context.history[metric].isnull().sum() == 3
-
-
-def test_metric_default(relay_server, wandb_init):
-    with relay_server() as relay:
-        run = wandb_init()
-        run_id = run.id
+@pytest.mark.parametrize("summary_type", [None, "copy"])
+def test_default_summary_type_is_last(wandb_backend_spy, summary_type):
+    with wandb.init() as run:
+        run.define_metric("*", summary=summary_type)
         run.log(dict(mystep=1, val=2))
         run.log(dict(mystep=2, val=8))
         run.log(dict(mystep=3, val=3))
         run.log(dict(val2=4))
         run.log(dict(val2=1))
-        run.finish()
 
-    summary = relay.context.get_run_summary(run_id)
+    with wandb_backend_spy.freeze() as snapshot:
+        summary = snapshot.summary(run_id=run.id)
+        assert summary["val"] == 3
+        assert summary["val2"] == 1
+        assert summary["mystep"] == 3
 
-    # by default, we use last value
-    assert summary["val"] == 3
-    assert summary["val2"] == 1
-    assert summary["mystep"] == 3
-    assert len(summary) == 3
 
-
-def test_metric_copy(relay_server, wandb_init):
-    with relay_server() as relay:
-        run = wandb_init()
-        run_id = run.id
-        run.define_metric("*", summary="copy")
-        run.log(dict(mystep=1, val=2))
-        run.log(dict(mystep=2, val=8))
-        run.log(dict(mystep=3, val=3))
-        run.log(dict(val2=4))
-        run.log(dict(val2=1))
-        run.finish()
-
-    summary = relay.context.get_run_summary(run_id)
-
-    assert summary["val"] == 3
-    assert summary["val2"] == 1
-    assert summary["mystep"] == 3
-    assert len(summary) == 3
-
-
-def test_metric_glob_none(relay_server, wandb_init):
-    with relay_server() as relay:
-        run = wandb_init()
-        run_id = run.id
+def test_summary_type_none(wandb_backend_spy):
+    with wandb.init() as run:
```

Comment:
we had some default settings values, maybe it is fine to ignore these, just take a look to confirm

Reply:
It sets the "name" setting, sets "console" to "off", and "save_code" to False; I think that's all fine to skip. If we want any of these to be defaults for all tests, it'll be cleaner to use autouse fixtures that set their corresponding env variables.